tiup: add bench.md and cluster.md (#2479)
pingcap/docs: reference/tools/tiup/bench.md

---
title: Stress Test TiDB Using TiUP Bench Component
summary: Learns how to stress test TiDB with TPC-C and TPC-H workloads using TiUP.
category: tools
---

# Stress Test TiDB Using TiUP Bench Component

> When you test the performance of a database, it is often required to stress test the database. To facilitate this, TiUP has integrated the bench component, which provides two workloads for stress testing: TPC-C and TPC-H. The commands and parameters are as follows:

Reviewer: ```suggestion
When you test the performance of a database, it is often required to stress test the database. To facilitate this, TiUP has integrated the bench component, which provides two workloads for stress testing: TPC-C and TPC-H. The commands and flags are as follows:
```

  -h, --help                help for /Users/joshua/.tiup/components/bench/v0.0.1/bench
  -H, --host string         Database host (default "127.0.0.1")
      --ignore-error        Ignore error when running workload
      --interval duration   Output interval time (default 10s)
      --isolation int       Isolation Level 0: Default, 1: ReadUncommitted, 
                            2: ReadCommitted, 3: WriteCommitted, 4: RepeatableRead, 
                            5: Snapshot, 6: Serializable, 7: Linerizable
      --max-procs int       runtime.GOMAXPROCS
  -p, --password string     Database password
  -P, --port int            Database port (default 4000)
      --pprof string        Address of pprof endpoint
      --silence             Don't print error when running workload
      --summary             Print summary TPM only, or also print current TPM when running workload
  -T, --threads int         Thread concurrency (default 16)
      --time duration       Total execution time (default 2562047h47m16.854775807s)
  -U, --user string         Database user (default "root")
```

The following sections describe how to run TPC-C and TPC-H tests using TiUP.

## Run TPC-C test using TiUP

> The TiUP bench component supports the following commands and parameters of the TPC-C test:

Reviewer: ```suggestion
The TiUP bench component supports the following commands and flags to run the TPC-C test:
```

```bash
Available Commands:
  check       Check data consistency for the workload
  cleanup     Cleanup data for the workload
  prepare     Prepare data for the workload
  run         Run workload

Flags:
      --check-all        Run all consistency checks
  -h, --help             help for tpcc
      --output string    Output directory for generating csv file when preparing data
      --parts int        Number to partition warehouses (default 1)
      --tables string    Specified tables for generating file, separated by ','. Valid only if output is set. If this flag is not set, generate all tables by default.
      --warehouses int   Number of warehouses (default 10)
```

### Test procedures

> 1. Create 4 repositories using 4 partitions with hash:

Reviewer: ```suggestion
1. Create 4 warehouses using 4 partitions via hash:
```
See the command for the term usage.

    ```shell
    tiup bench tpcc --warehouses 4 --parts 4 prepare
    ```

2. Run the TPC-C test:

    {{< copyable "shell-regular" >}}

    ```shell
    tiup bench tpcc --warehouses 4 run
    ```

3. Clean up data:

    {{< copyable "shell-regular" >}}

    ```shell
    tiup bench tpcc --warehouses 4 cleanup
    ```

4. Check the consistency:

    {{< copyable "shell-regular" >}}

    ```shell
    tiup bench tpcc --warehouses 4 check
    ```

5. Generate the CSV file:

    {{< copyable "shell-regular" >}}

    ```shell
    tiup bench tpcc --warehouses 4 prepare --output data
    ```

6. Generate the CSV file for the specified table:

    {{< copyable "shell-regular" >}}

    ```shell
    tiup bench tpcc --warehouses 4 prepare --output data --tables history,orders
    ```

7. Enable pprof:

    {{< copyable "shell-regular" >}}

    ```shell
    tiup bench tpcc --warehouses 4 prepare --output data --pprof :10111
    ```

## Run TPC-H test using TiUP

> The TiUP bench component supports the following commands and parameters of the TPC-P test:

Reviewer: ```suggestion
The TiUP bench component supports the following commands and parameters to run the TPC-H test:
```
